apps/cowswap-frontend/src/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -72,7 +72,11 @@ export function useSetupTradeState(): void {
         }
       }
 
-      tradeNavigate(providerChainId, getDefaultTradeRawState(providerChainId))
+      // Adding timeout of 500ms to avoid running into infinity loop of updating provider and url state.
+      // issue GH : https://github.com/cowprotocol/cowswap/issues/4734
+      setTimeout(() => {
+        tradeNavigate(providerChainId, getDefaultTradeRawState(providerChainId))
+      }, 500)
```

Comment:
It seems fragile to rely on a simple timeout. 

Would debouncing the whole callback work?
https://github.com/cowprotocol/cowswap/blob/use-env-for-cms-endpoint/libs/common-utils/src/misc.ts#L34

Reply:
Great!, Didn't realise there was a debounce implementation already. Update code with debouncing the `switchNetworkInWallet` function 👍 

The reason for debouncing  `switchNetworkInWallet` instead of `tradeNavigate` is to address the issue that @elena-zh  comment.